src/runner.rs

```diff
@@ -128,3 +128,104 @@ impl Runner for FirefoxRunner {
         Ok(self.ret_code)
     }
 }
+
+pub fn find_binary(name: &str) -> Option<PathBuf> {
+    env::var("PATH")
+        .ok()
+        .and_then(|path_env| {
+            for mut path in env::split_paths(&*path_env) {
+                path.push(name);
+                if path.exists() {
+                    return Some(path)
+                }
+            }
+            None
+        })
+}
+
+#[cfg(target_os = "linux")]
+pub mod platform {
+    use super::find_binary;
+    use std::path::PathBuf;
+
+    pub fn firefox_default_path() -> Option<PathBuf> {
+        find_binary("firefox")
+    }
+}
+
+#[cfg(target_os = "macos")]
+pub mod platform {
+    use super::find_binary;
+    use std::env;
+    use std::path::PathBuf;
+
+    pub fn firefox_default_path() -> Option<PathBuf> {
+        let path = PathBuf::from("/Applications/Firefox.app/Contents/MacOS/firefox-bin");
+        if path.exists() {
+            return Some(path.to_path_buf())
+        };
+        let home_path = env::home_dir()
+            .map(|x| x.join("Applications/Firefox.app/Contents/MacOS/firefox-bin"));
+        if let Some(path) = home_path {
+            if path.exists() {
+                return Some(path)
+            }
+        };
```

Comment:
The above code could be generalised to a vector and iterated over, so that it could support finding it in multiple locations.